server/internal/usecase/interactor/scene.go

```diff
@@ -557,6 +557,113 @@ func (i *Scene) RemoveCluster(ctx context.Context, sceneID id.SceneID, clusterID
 	return s, nil
 }
 
+func (i *Scene) AddStyle(ctx context.Context, param interfaces.AddStyleInput, operator *usecase.Operator) (*scene.Scene, *scene.Style, error) {
+	tx, err := i.transaction.Begin(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ctx = tx.Context()
+	defer func() {
+		if err2 := tx.End(ctx); err == nil && err2 != nil {
+			err = err2
+		}
+	}()
+
+	s, err := i.sceneRepo.FindByID(ctx, param.SceneID)
+	if err != nil {
+		return nil, nil, err
+	}
+	if err := i.CanWriteWorkspace(s.Workspace(), operator); err != nil {
+		return nil, nil, err
+	}
+
+	cid := id.NewStyleID()
+	style, err := scene.NewStyle(cid, *param.Name, param.Value)
```

Comment:
its not good to consider `param.Name` as its always not nil, its better to check it or use `string` instead of `*string`